src/linalg/naive/dense_matrix.rs

```diff
@@ -444,6 +445,38 @@ impl<T: RealNumber> LUDecomposableMatrix<T> for DenseMatrix<T> {}
 
 impl<T: RealNumber> CholeskyDecomposableMatrix<T> for DenseMatrix<T> {}
 
+impl<T: RealNumber> HighOrderOperations<T> for DenseMatrix<T> {
+    fn ab(&self, a_transpose: bool, b: &Self, b_transpose: bool) -> Self {
```

Comment:
Is there a specific reason to not use the default implementation of `fn ab`?

Reply:
the default implementation is less efficient because you run 2 for loops every transpose. In the worst case scenario, when you have to do 2 transposes you will end up doing 6 for loops. In addition to that every transpose creates a new 2d array. 
Bundling matrix operations together can improve performance since you can do many transformations in the same iteration. Take a look at https://nalgebra.org/quick_reference/#blas-operations